pubsub/google/cloud/pubsub_v1/publisher/batch/thread.py

```diff
@@ -0,0 +1,385 @@
+                considered immutable once the batch has been opened.
+        """
+        return self._settings
+
+    @property
+    def size(self):
+        """Return the total size of all of the messages currently in the batch.
+
+        Returns:
+            int: The total size of all of the messages currently
+                 in the batch, in bytes.
+        """
+        return self._size
+
+    @property
+    def status(self):
+        """Return the status of this batch.
+
+        Returns:
+            str: The status of this batch. All statuses are human-readable,
+                all-lowercase strings.
+        """
+        return self._status
+
+    def commit(self):
+        """Actually publish all of the messages on the active batch.
+
+        This synchronously sets the batch status to in-flight, and then opens
+        a new thread, which handles actually sending the messages to Pub/Sub.
+
+        .. note::
+
+            This method is non-blocking. It opens a new thread, which calls
+            :meth:`_commit`, which does block.
+        """
+        # Set the status to in-flight synchronously, to ensure that
+        # this batch will necessarily not accept new messages.
+        #
+        # Yes, this is repeated in `_commit`, because that method is called
+        # directly by `monitor`.
+        self._status = 'in-flight'
+
+        # Start a new thread to actually handle the commit.
+        commit_thread = threading.Thread(target=self._commit)
+        commit_thread.start()
+
+    def _commit(self):
+        """Actually publish all of the messages on the active batch.
+
+        This moves the batch out from being the active batch to an in-flight
+        batch on the publisher, and then the batch is discarded upon
+        completion.
+
+        .. note::
+
+            This method blocks. The :meth:`commit` method is the non-blocking
+            version, which calls this one.
+        """
+        # Update the status.
+        self._status = 'in-flight'
+
+        # Begin the request to publish these messages.
+        if len(self._messages) == 0:
+            return
+        response = self.client.api.publish(
+            self._topic,
+            self.messages,
+        )
+
+        # We got a response from Pub/Sub; denote that we are processing.
+        self._status = 'processing results'
+
+        # Sanity check: If the number of message IDs is not equal to the
+        # number of futures I have, then something went wrong.
+        if len(response.message_ids) != len(self._futures):
+            raise exceptions.PublishError(
+                'Some messages were not successfully published.',
+            )
+
+        # Iterate over the futures on the queue and return the response IDs.
+        # We are trusting that there is a 1:1 mapping, and raise an exception
+        # if not.
+        self._status = self.Status.SUCCESS
+        for message_id, fut in zip(response.message_ids, self._futures):
```

Comment:
I don't see a `future` import? At any rate you can use `import google.cloud.future` and reference the fullly-qualified module name for disambiguation.